react-client/src/components/Pet/PetCard.tsx

```diff
@@ -1,40 +1,58 @@
 import { classNames } from "~/util/styleUtil";
 import { Card, Icon } from "../design-system";
+import useCardBase, { UseCardBase } from "../design-system/card/useCardBase";
+import { isBoolean } from "lodash";
+import { ReactNode } from "react";
 
-type PetCardProps = {
-  name: string;
+type PetCardProps = UseCardBase & {
+  children: ReactNode;
+  hasShadow?: boolean;
   img?: string;
-  isProtected: boolean;
+  isProtected?: boolean;
+  name: string;
 };
 
-export const PetCard = ({ name, img, isProtected, ...props }: PetCardProps) => {
+export const PetCard = ({
+  children,
+  hasShadow,
+  img,
+  isProtected,
+  name,
+  variant,
+  ...props
+}: PetCardProps) => {
+  const { className: baseClassName } = useCardBase({
+    variant,
+  });
+  const isSimpleCard = isBoolean(isProtected);
```

Comment:
You don't need that dependency since you can easily transform it on a boolean with double exclamation mark;
```suggestion
  const isSimpleCard = !!isProtected;
```